src/vhip/mlmodel/gene_features.py

```diff
@@ -352,3 +354,38 @@ def RSCU(
                         synonymous_total_count / len(synonymous_codons)
                     )  # expected frequency of current codon given all assumption all synonymous codons are equally likely to encode the aa
                     self.RSCU_dict[codon] = self.codon_dict[codon] / expected_frequency
+
+
+class CodonBiasComparison:
+    """Class for calculating codon bias similarity between a virus GeneSet and a host GeneSet.
+
+    Args:
+        host_dict: Dictionary of codons/amino acids and their counts/frequencies/RSCUs in a host GeneSet.
+        virus_dict: Dictionary of codons/amino acids and their counts/frequencies/RSCUs in a virus GeneSet.
+    """
+
+    def __init__(
+        self,
+        host_dict: Union[dict[str, int], dict[str, float]],
```

Comment:
if the typing is an Union, then it is not a dictionary right? If so, I'd suggest changing the variable name to avoid confusion (and if I am wrong please let me know!)

Reply:
The Union just indicates that the input can be either of the two dictionary types specified. I believe I initially tried using an or or | operator, but pyright was unhappy with that. I think this accomplishes the same thing, though.

Reply:
ahhh I learned something new today! thank you for clarifying